### DISCUSSION TITLE
What are the use cases for methods with BooleanSupplier parameter?

### DISCUSSION THREAD

**Discussion by Marcono1234** [junit-team/junit-framework#2552] [Q&A] [answered]
There are a few methods in `Assertions` and `Assumptions` which have a `java.util.function.BooleanSupplier` as parameter, e.g. [`Assertions.assertTrue(BooleanSupplier)`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertTrue(java.util.function.BooleanSupplier)). They were introduced by ad2b5fec1128112b7d68fdaa9b4212099f6d9092 and b8d922994b76478e2024b3087f2fe8e20679fc1f.

What is the use case for these methods?
Given that JUnit has to always get the boolean value from the supplier and that JUnit is apparently not enhancing any exceptions thrown by the supplier (or when the supplier is `null`), the only reasonable use case I can think of are classes which actually implement `BooleanSupplier`, though I guess that is rather rare. In all other use cases it would be easier to directly provide the boolean value.

**Comment by sbrannen** [comment, marked as answer]
> the only reasonable use case I can think of are classes which actually implement `BooleanSupplier`

That is one category of use cases: types that implement `BooleanSupplier`.

Another category of use cases includes methods that return a `BooleanSupplier`.

> In all other use cases it would be easier to directly provide the boolean value.

Yes, if you already have the `boolean` or `Boolean`, there is no need to wrap it in a `BooleanSupplier`.

**Reply by Marcono1234** [reply]
Thank you for the answer. Using a type implementing `BooleanSupplier` directly as argument of an assertion method might reduce readability in some cases (especially when the name of the type does not indicate that it can supply `boolean` values), but I guess there are also cases then where it might be useful to have such assertion methods.
